stack/stack.go

```diff
@@ -0,0 +1,252 @@
+		OsxXcode112X,
+		OsxXcode111X,
+		OsxXcode110X,
+		OsxXcode101X,
+		OsxXcode100X,
+		OsxXcode94X,
+		OsxXcode83X,
+		OsxXcodeEdge,
+	},
+	Android: {
+		LinuxDockerAndroid,
+		LinuxDockerAndroidLts,
+		OsxVs4MACStable,
+		OsxVs4MACBeta,
+		OsxVs4MACPreviousStable,
+		OsxXcode112X,
+		OsxXcode111X,
+		OsxXcode110X,
+		OsxXcode102X,
+		OsxXcode101X,
+		OsxXcode100X,
+		OsxXcode94X,
+		OsxXcode83X,
+		OsxXcodeEdge,
+	},
+	MacOS: {
+		OsxXcode102X,
+		OsxVs4MACStable,
+		OsxVs4MACBeta,
+		OsxVs4MACPreviousStable,
+		OsxXcode112X,
+		OsxXcode111X,
+		OsxXcode110X,
+		OsxXcode101X,
+		OsxXcode100X,
+		OsxXcode94X,
+		OsxXcode83X,
+		OsxXcodeEdge},
+	Ios: {
+		OsxXcode102X,
+		OsxVs4MACStable,
+		OsxVs4MACBeta,
+		OsxVs4MACPreviousStable,
+		OsxXcode112X,
+		OsxXcode111X,
+		OsxXcode110X,
+		OsxXcode101X,
+		OsxXcode100X,
+		OsxXcode94X,
+		OsxXcode83X,
+		OsxXcodeEdge},
+}
+
+// Stack defines a Stack with it's name.
+type Stack string
+
+const (
+	// LinuxDockerAndroidLts the stack for linux-docker-android-lts.
+	LinuxDockerAndroidLts Stack = "linux-docker-android-lts"
+	// LinuxDockerAndroid the stack for linux-docker-android.
+	LinuxDockerAndroid Stack = "linux-docker-android"
+	// OsxVs4MACBeta the stack for osx-vs4mac-beta.
+	OsxVs4MACBeta Stack = "osx-vs4mac-beta"
+	// OsxVs4MACPreviousStable the stack for osx-vs4mac-previous-stable.
+	OsxVs4MACPreviousStable Stack = "osx-vs4mac-previous-stable"
+	// OsxVs4MACStable the stack for osx-vs4mac-stable.
+	OsxVs4MACStable Stack = "osx-vs4mac-stable"
+	// OsxXcode112X the stack for osx-xcode-11.2.x.
+	OsxXcode112X Stack = "osx-xcode-11.2.x"
+	// OsxXcode111X the stack for osx-xcode-11.1.x.
+	OsxXcode111X Stack = "osx-xcode-11.1.x"
+	// OsxXcode110X the stack for osx-xcode-11.0.x.
+	OsxXcode110X Stack = "osx-xcode-11.0.x"
+	// OsxXcode102X the stack for osx-xcode-10.2.x.
+	OsxXcode102X Stack = "osx-xcode-10.2.x"
+	// OsxXcode101X the stack for osx-xcode-10.1.x.
+	OsxXcode101X Stack = "osx-xcode-10.1.x"
+	// OsxXcode100X the stack for osx-xcode-10.0.x.
+	OsxXcode100X Stack = "osx-xcode-10.0.x"
+	// OsxXcode94X the stack for osx-xcode-9.4.x
+	OsxXcode94X Stack = "osx-xcode-9.4.x"
+	// OsxXcode83X the stack for osx-xcode-8.3.x.
+	OsxXcode83X Stack = "osx-xcode-8.3.x"
+	// OsxXcodeEdge the stack for osx-xcode-edge.
+	OsxXcodeEdge Stack = "osx-xcode-edge"
+)
+
+// Stacks is the array of the all available stacks.
+var Stacks = []Stack{
+	LinuxDockerAndroidLts, LinuxDockerAndroid, OsxVs4MACBeta, OsxVs4MACPreviousStable, OsxXcode112X,
+	OsxXcode111X, OsxXcode110X, OsxXcode102X, OsxXcode101X, OsxXcode100X, OsxXcode94X,
+	OsxXcode83X, OsxXcodeEdge,
+}
+
+// ParseStack gets the given stack from a string.
+func ParseStack(s string) (Stack, error) {
+	for _, stack := range Stacks {
+		if string(stack) == s {
+			return stack, nil
+		}
+	}
+	return "", fmt.Errorf("could not find stack %s", s)
+}
+
+// StringValue returns the string value of the given Stack.
+func (s Stack) StringValue() string {
```

Comment:
This name should be `String` by Go's built-in.

Reply:
Renamed.